### ISSUE TITLE
There is an issue with reading the load file

### ISSUE THREAD

**Issue by Cola-Fsm** [decisionintelligence/DUET#5]
Why is there an error in the parameter handling of this code?

**Comment by qiu69** [comment]
Thank you for your question. I'm sorry I didn't fully understand your meaning. Would it be convenient for you to take a screenshot.

**Comment by YEKAI-2022** [comment]
![image](https://github.com/user-attachments/assets/72e3e9b9-ba20-42ec-b89d-166110d4ea50)Hello, may I ask why this error is being reported?

**Comment by qiu69** [comment]
@YEKAI-2022 Can you tell me your running script？

**Comment by YEKAI-2022** [comment]
> @YEKAI-2022 Can you tell me your running script？
在pycharm中的命令
--config-path
"rolling_forecast_config.json"
--data-name-list
"dataset/ETT-small/ETTh1.csv"
--strategy-args
'{"horizon":96}'
--model-name
"duet.DUET"
--model-hyper-params
'{"batch_size":
32,
"d_ff":
512,
"d_model":
512,
"dropout":
0.5,
"fc_dropout":
0.1,"n_heads":
1,
"e_layers":
1,
"lr":
0.0005,
"horizon":
96,
"seq_len":
512,
"factor":
3,
"lradj":
"type1",
"loss":
"MAE",
"num_experts":
2,
"k":
1,
"patch_len":
48,
"patience":
5,
"num_epochs":
100,
"CI":
1}'
--gpus
0
--num-workers
0
--timeout
60000
--save-path
"ETTh1/DUET"

**Comment by Cola-Fsm** [comment]
> @YEKAI-2022 Can you tell me your running script？

I made the same mistake as him, the script seems to have used the wrong JSON format, using the script you provided.

**Comment by qiu69** [comment]
<img width="931" alt="image" src="https://github.com/user-attachments/assets/3cb4cd7e-d119-43d0-9383-59fa9791cecd" />

please refer to: https://github.com/decisionintelligence/DUET, FAQ: Q1

**Comment by YEKAI-2022** [comment]
![image](https://github.com/user-attachments/assets/dfccb2d3-f6e4-4fa8-9d40-5572471d103e)
我下好了数据集为什么路径错了，--data-name-list"E:\\AI\\Time series analysis\\DUET-main\\DUET-main\\dataset\\ETT-small\\ETTh1.csv"

**Comment by YEKAI-2022** [comment]
请问下载好的数据集的路径该怎么指定呀

**Comment by Cola-Fsm** [comment]
> 请问下载好的数据集的路径该怎么指定呀
相同的问题 好像没有把路径传过去

**Comment by qiu69** [comment]
您好，请在DUET的readme里面下载DUET的数据集。然后只要指定数据集的名称就行，不用写路径。

**Comment by qiu69** [comment]
<img width="242" alt="image" src="https://github.com/user-attachments/assets/d6ec7550-ae3b-4807-9b31-161c86cb5ac7" />

**Comment by qiu69** [comment]
请问问题得到解决了嘛？

**Comment by Cola-Fsm** [comment]
> 请问问题得到解决了嘛？

能够训练 但是结束的时候报错了
![QQ_1736040631011](https://github.com/user-attachments/assets/96fd9f4c-4eac-4a37-a8fa-590ea159d422)

**Comment by qiu69** [comment]
请问跑出结果了吗，以及您运行的脚本方便发一下吗？

**Comment by Cola-Fsm** [comment]
python ./scripts/run_benchmark.py --config-path "rolling_forecast_config.json" --data-name-list "ILI.csv" --strategy-args {\"horizon\":24} --model-name "duet.DUET" --model-hyper-params {\"batch_size\":8,\"dropout\":0.15,\"fc_dropout\":0,\"d_ff\":1024,\"d_model\":128,\"n_heads\":1,\"e_layers\":2,\"lr\":0.0005,\"horizon\":24,\"seq_len\":104,\"factor\":3,\"lradj\":\"type1\",\"loss\":\"MAE\",\"num_experts\":2,\"k\":2,\"patch_len\":48,\"patience\":5,\"num_epochs\":100,\"CI\":1} --gpus 0 --num-workers 1 --timeout 60000 --save-path "ILI/DUET"
![QQ_1736041032435](https://github.com/user-attachments/assets/a75c223a-03c3-4e3a-b25a-e111c7469314)
生成了一个压缩包

**Comment by qiu69** [comment]
压缩包里面看mae_norm和mse_norm这两列就行，这也是论文中汇报的对应mae和mse结果，指在归一化后的数据集以及预测结果上面计算误差指标。压缩包里面的mae和mse对应是返归一化后计算误差指标。

**Comment by qiu69** [comment]
直接运行：sh ./scripts/multivariate_forecast/ETTh1_script/DUET.sh。看看是否报错。以及您方便加我微信吗，qxf780310.

**Comment by YEKAI-2022** [comment]
> 您好，请在DUET的readme里面下载DUET的数据集。然后只要指定数据集的名称就行，不用写路径。

![image](https://github.com/user-attachments/assets/6c89ca20-24d1-4f7e-817a-7e1b1d975cd1)
你好，我下载数据集后还是出现这个问题

**Comment by liukaiyueyuo** [comment]
请问：No such file or directory: './dataset/forecasting/FORECAST_META.csv'。这个FORECAST_META.csv是啥文件？哪里下载？ 为啥报错？

**Comment by liukaiyueyuo** [comment]
@qiu69 请问：No such file or directory: './dataset/forecasting/FORECAST_META.csv'。这个FORECAST_META.csv是啥文件？哪里下载？ 为啥报错？

**Comment by qiu69** [comment]
@liukaiyueyuo 您好，readme里面有下载数据集的链接，FORECAST_META.csv这个文件已经被包含在内了。
<img width="876" alt="image" src="https://github.com/user-attachments/assets/02fd3750-35d7-45dc-8ee7-21188ae6d121" />

**Comment by liruyungitplayer** [comment]
老师您好，我在pycharm终端运行脚本时，参数也报错了，并且找了很久也找不到原因，老师您能帮我看一下吗

![Image](https://github.com/user-attachments/assets/17df6424-e608-4458-800f-dfc76468b16b)

**Comment by liruyungitplayer** [comment]
> 老师您好，我在pycharm终端运行脚本时，参数也报错了，并且找了很久也找不到原因，老师您能帮我看一下吗
> 
> ![Image](https://github.com/user-attachments/assets/17df6424-e608-4458-800f-dfc76468b16b)
已经改好了，在大括号外边加上单引号就可以了

**Comment by sdGundamf** [comment]
命令行很难用ε=(´ο｀*)))唉

**Comment by qiu69** [comment]
@sdGundamf 您好，后面我们会把scripts里面的脚本改成类似如下形式，您看是否会方便很多！

<img width="367" alt="Image" src="https://github.com/user-attachments/assets/82ba6d9b-41fe-4f4d-ad69-1707f9e894c9" />

**Comment by omenkfjsjjssc** [comment]
> [@sdGundamf](https://github.com/sdGundamf)您好，后面我们把脚本里面的脚本改成如下类似的形式，您看看是否会方便很多！
> 
> <img alt="图像" width="367" src="https://private-user-images.githubusercontent.com/82441982/424415706-82ba6d9b-41fe-4f4d-ad69-1707f9e894c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ3ODY1ODYsIm5iZiI6MTc0NDc4NjI4NiwicGF0aCI6Ii84MjQ0MTk4Mi80MjQ0MTU3MDYtODJiYTZkOWItNDFmZS00ZjRkLWFkNjktMTcwN2Y5ZTg5NGM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE2VDA2NTEyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmYmM4ZTBiNjUzMDY4ZTc5ZjZhYTRjMTM5ODZkYjZkZjdiNzI0Nzg5MDkyMDJhM2MxNjM4Y2MwZWZmZDRiYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.x9bGPtYPkKnf6AGVf8zbHORHCcI4feMSyFdXNzS2Vyc">

Could you please send this file using the new script to my email? Thank you very much Email：183909450@qq.com

**Comment by omenkfjsjjssc** [comment]
I can't find the code corresponding to each module of your model and where is the main model, can you explain

**Comment by qiu69** [comment]
Please see: https://github.com/decisionintelligence/DUET/blob/main/ts_benchmark/baselines/duet/models/duet_model.py

**Comment by Jerry-a-rookie** [comment]
您好，我在实现您的结果时数据路径总是出错，能否提供详细的数据路径

**Comment by qiu69** [comment]
@Jerry-a-rookie 请问报什么错呢，请问您有下载readme里面我们提供的数据集吗？

**Comment by Jerry-a-rookie** [comment]
谢谢您的回复，我已经成功运行。还想请教一下您超参数调整的问题



---Original---
From: "Xiangfei ***@***.***&gt;
Date: Sun, Jul 6, 2025 13:50 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [decisionintelligence/DUET] There is an issue with reading theload file (Issue #5)


qiu69 left a comment (decisionintelligence/DUET#5)
 
@Jerry-a-rookie 请问报什么错呢，请问您有下载readme里面我们提供的数据集吗？
